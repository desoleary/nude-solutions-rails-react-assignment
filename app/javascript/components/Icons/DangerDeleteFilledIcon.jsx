import { DeleteFilled as DeleteFilledAntd } from '@ant-design/icons/lib/icons';
import React from 'react';

import { string, shape } from 'prop-types';

const DangerDeleteFilledIcon = (props) => <DeleteFilledAntd {...props} />;

DangerDeleteFilledIcon.defaultProps = {
  style: { color: 'red' }
};

DangerDeleteFilledIcon.propTypes = {
  style: shape({ color: string })
};

export default DangerDeleteFilledIcon;
