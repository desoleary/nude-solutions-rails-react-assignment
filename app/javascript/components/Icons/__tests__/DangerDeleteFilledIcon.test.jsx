import { DeleteFilled } from '@ant-design/icons/lib/icons';

import { mount } from 'enzyme';
import React from 'react';
import DangerDeleteFilledIcon from '../DangerDeleteFilledIcon';

describe('with shallow rendering', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<DangerDeleteFilledIcon />);
  });

  it('compares snapshot', () => {
    expect(wrapper.length).to.eql(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("passes props correctly to Antd's DeleteFilled Icon", () => {
    const antdDeleteFilledProps = wrapper.find(DeleteFilled).props();

    expect(antdDeleteFilledProps.style).to.eql({ color: 'red' });
  });
});
