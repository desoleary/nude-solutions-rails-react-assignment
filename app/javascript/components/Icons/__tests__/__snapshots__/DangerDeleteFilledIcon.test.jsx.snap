// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with shallow rendering compares snapshot 1`] = `
<DangerDeleteFilledIcon
  style={
    Object {
      "color": "red",
    }
  }
>
  <ForwardRef(DeleteFilled)
    style={
      Object {
        "color": "red",
      }
    }
  >
    <AntdIcon
      icon={
        Object {
          "icon": Object {
            "attrs": Object {
              "focusable": "false",
              "viewBox": "64 64 896 896",
            },
            "children": Array [
              Object {
                "attrs": Object {
                  "d": "M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z",
                },
                "tag": "path",
              },
            ],
            "tag": "svg",
          },
          "name": "delete",
          "theme": "filled",
        }
      }
      style={
        Object {
          "color": "red",
        }
      }
    >
      <span
        aria-label="delete"
        className="anticon anticon-delete"
        role="img"
        style={
          Object {
            "color": "red",
          }
        }
      >
        <IconReact
          className=""
          icon={
            Object {
              "icon": Object {
                "attrs": Object {
                  "focusable": "false",
                  "viewBox": "64 64 896 896",
                },
                "children": Array [
                  Object {
                    "attrs": Object {
                      "d": "M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z",
                    },
                    "tag": "path",
                  },
                ],
                "tag": "svg",
              },
              "name": "delete",
              "theme": "filled",
            }
          }
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="delete"
            fill="currentColor"
            focusable="false"
            height="1em"
            key="svg-delete"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
              key="svg-delete-svg-0"
            />
          </svg>
        </IconReact>
      </span>
    </AntdIcon>
  </ForwardRef(DeleteFilled)>
</DangerDeleteFilledIcon>
`;
